test/unittest/test_utils.py

```diff
@@ -74,21 +74,13 @@ def test_open_for_write_file_exists(self):
         with mock.patch('io.open',
                         mock.mock_open(),
                         create=True) as mock_open, \
-             mock.patch('os.path.exists', return_value=True):
+             mock.patch.object(
+                utils.pathlib.Path,
+                'mkdir',
+                return_value=True) as mock_create:
             utils._open_for_write(filename)
 
-            mock_open.assert_called_once_with(filename, 'w')
-
-    def test_open_for_write_file_does_not_exist(self):
-        filename = 'foo/bar/new_file'
-        with mock.patch('io.open',
-                        mock.mock_open(),
-                        create=True) as mock_open, \
-             mock.patch('os.path.exists', return_value=False), \
-             mock.patch('os.makedirs') as mock_makedirs:
-            utils._open_for_write(filename)
-
-            mock_makedirs.assert_called_once_with(os.path.dirname(filename))
+            mock_create.assert_called_once()
```

Comment:
can you use `assert_called_once_with(os.path.dirname(filename), parents=True, exists_ok=True)`

Reply:
```
Actual call: mkdir(exist_ok=True, parents=True)
>>  raise AssertionError("Expected call: mkdir('foo/bar', exists_ok=True, parents=True)\nActual call: mkdir(exist_ok=True, parents=True)",)
```

The Path object uses internal state to know what the path is. It's not passed to makedir.

Reply:
I could write an assert that checks that the internal path matches the expected one. I'm not sure how to do that effectively without mocking the Path object and that seems more complex then it needs to be.